grant/projects/offchain-subnames.md

```diff
@@ -0,0 +1,145 @@
+
+This project addresses the need for efficient subdomain management within the Darwinia ecosystem, as outlined in the RFP (https://github.com/ringecosystem/collaboration/issues/61), and demonstrates the potential for cross-chain interoperability in domain name services.
+
+### This project will benefit the Darwinia ecosystem in the following ways:
+
+1. It will provide a more efficient and cost-effective way to manage subdomains on the Darwinia Chain, reducing the need for users to pay high gas fees on Ethereum.
+2. It will enhance the usability of the 'darwinia.eth' domain by allowing users to register, update, and resolve subnames with reduced costs and improved performance.
+3. It will demonstrate the potential for cross-chain interoperability in domain name services, opening up new possibilities for collaboration and innovation within the Darwinia ecosystem.
+
+## Team
+
+- Aki Wu([wuminzhe](https://github.com/wuminzhe))  
+  A full stack developer focused on the web3 space. He has been working on darwinia ecosystem for a long time.
+
+- Echo([Echo](https://github.com/hujw77))  
+  Contracts developer from Itering.
+
+- Bear([Bear](https://github.com/boundless-forest))
+  Project Manager from Itering. 
+  
+- Yuqi(yuqi.liu@itering.io)
+  UI Designer from Itering.
+
+## Team Code Repos
+
+- https://github.com/subnames/subnames-contracts
+  The repo for the subnames contracts based on [basenames](https://github.com/base-org/basenames).
+
+- https://github.com/subnames/offchain-resolver  
+  1. the offchain gateway.
+  2. the offchain resolve client.
+  3. based on ensdomains's [offchain-resolver](https://github.com/ensdomains/offchain-resolver).
+
+- https://github.com/wuminzhe/subnames-ui
+  Frontend User Interface.
+
+## Development Roadmap
+
+### Milestone 1
+
+Estimated Duration: 2 weeks
+
+Cost: 1000 USDT
+
+Address: 0x2Da8ccfe0dD2165B8d939eaBf4E3697C4Adb6FDd
+
+#### Tasks:
+
+1. Configure the contracts to support "ringdao.eth" domain for testing. 
+2. Deploy contracts to testnet(Darwinia Koi).
+3. Implement and deploy the offchain gateway.
+4. Implement the Wallet Connect button including customizing it to support subname.
+5. Support registering subname and fee calculation.
+6. Support extend/renew.
+7. Support reverse resolution(address to domain).
+
+Deliverables:
+
+- The contracts source code.
+- The gateway source code.
+- A deployed gateway service targetting Darwinia Koi.
+- The deployed contracts addresses on Darwinia Koi.
+- The frontend UI for wallet connecting, registering and extend/renew on Darwinia Koi.
+
+** Note: this milestone has done. **
+
+### Milestone 2
+
+Estimated Duration: 2 weeks
+
+Cost: 500 USDT
+
+Address: 0x2Da8ccfe0dD2165B8d939eaBf4E3697C4Adb6FDd
+
+#### Tasks:
+
+1. Configure the contracts to support "darwinia.eth" domain. 
```

Comment:
Requesting to include "transferring the ownership of the smart contract to RingDAO" in the deliveries of the grant, to eliminate the vendor lock-in risk.